build/teamcity/cockroach/nightlies/microbenchmark_weekly.sh

```diff
@@ -75,6 +76,7 @@ done
 
 # Check if the baseline cache exists and copy it to the output directory.
 baseline_cache_path="gs://$BENCH_BUCKET/cache/$GCE_MACHINE_TYPE/$SANITIZED_BENCH_PACKAGE/${sha_arr[1]}"
+baseline_commit=${sha_arr[1]}
```

Comment:
It took me a while to understand the `unset` call below -- it's a way to signal that we used cached baseline results. But then we erase the baseline SHA, which we now have to copy over here because we actually need it.

Maybe we can signal that we used cached results using some kind of boolean variable instead of array deletion? Bonus for then keeping the symmetry between `sha_arr` and `name_arr`.

Reply:
True, the logic here is a bit befuddled. To remedy this I opted for a separate array that needs to be built and run, vs. an array to reference the SHAs we're comparing. I tried a boolean, but it felt cleaner using another array.